### PR TITLE
Support cache host parameter

### PR DESCRIPTION
**Pull request by SerhiiNahornyi**

### CODE REVIEW

File: src/main/java/org/prebid/cache/handlers/GetCacheHandler.java [hunk 1]
```diff
@@ -51,32 +78,108 @@ public GetCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposito
...
+        return request.queryParam(ID_KEY).map(id -> fetch(request, id, timerContext)).orElseGet(() -> {
             val responseMono = ErrorHandler.createInvalidParameters();
             return finalizeResult(responseMono, request, timerContext);
         });
+    }
+
+    private Mono<ServerResponse> fetch(final ServerRequest request,
+                                       final String id,
+                                       final Timer.Context timerContext) {
+        val cacheUrl = resolveCacheUrl(request);
```
Comment (And1sS) [inline src/main/java/org/prebid/cache/handlers/GetCacheHandler.java:89]: Please, replace val wih final var, check for similar occurences.

File: src/main/java/org/prebid/cache/handlers/GetCacheHandler.java [hunk 2]
```diff
@@ -51,32 +78,108 @@ public GetCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposito
...
+
+        val responseMono = StringUtils.contains(cacheUrl, config.getAllowedProxyHost())
+                || StringUtils.contains(cacheUrl, LOCALHOST)
```
Comment (And1sS) [inline src/main/java/org/prebid/cache/handlers/GetCacheHandler.java:92]: Can be simplified to  `StringUtils.containsAny(cacheUrl, config.getAllowedProxyHost(), LOCALHOST)`.

File: src/main/java/org/prebid/cache/handlers/GetCacheHandler.java [hunk 3]
```diff
@@ -51,32 +78,108 @@ public GetCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposito
...
+    private void validateServerResponse(final ClientResponse clientResponse,
+                                        final SynchronousSink<ClientResponse> sink) {
+        if (HttpStatus.OK.equals(clientResponse.statusCode())) {
+            metricsRecorder.getProxySuccess().mark();
+            sink.next(clientResponse);
+            return;
+        }
+        sink.error(new BadRequestException(
+                String.format("Unexpected status code: '%s', in response from remote server",
+                        clientResponse.statusCode())));
```
Comment (And1sS) [inline src/main/java/org/prebid/cache/handlers/GetCacheHandler.java]: Pls, consider replacing `String.format` with just `+` operator

File: src/main/java/org/prebid/cache/handlers/GetCacheHandler.java [hunk 4]
```diff
@@ -51,32 +78,108 @@ public GetCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposito
...
+    }
+
+    private static Mono<ServerResponse> fromClientResponse(final ClientResponse clientResponse) {
+        return ServerResponse.status(clientResponse.statusCode())
+                .headers(headerConsumer -> clientResponse.headers().asHttpHeaders().forEach(headerConsumer::addAll))
+                .body(clientResponse.bodyToMono(String.class), String.class);
+    }
 
+    private Mono<ServerResponse> processRequest(final ServerRequest request, final String keyIdParam) {
+        val normalizedId = String.format("%s%s", config.getPrefix(), keyIdParam);
```
Comment (And1sS) [inline src/main/java/org/prebid/cache/handlers/GetCacheHandler.java:153]: Same here

File: src/main/java/org/prebid/cache/handlers/GetCacheHandler.java [hunk 5]
```diff
@@ -51,32 +78,108 @@ public GetCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposito
...
+                        metricsRecorder.markMeterForTag(this.metricTagPrefix,
+                                MetricsRecorder.MeasurementTag.XML);
+                        return builder.createResponseMono(request,
+                                MediaType.APPLICATION_XML,
+                                wrapper);
+                    } else {
+                        // unhandled media type
+                        return Mono.error(new UnsupportedMediaTypeException(UNSUPPORTED_MEDIATYPE));
+                    }
+                })
```
Comment (And1sS) [inline src/main/java/org/prebid/cache/handlers/GetCacheHandler.java:178]: Please, extract to separate method. Smth like `wrapperToServerResponse`, or `fromWrapperResponse`

File: src/test/java/org/prebid/cache/handlers/PostCacheHandlerTests.java [hunk 6]
```diff
@@ -226,12 +232,12 @@ void testUUIDDuplication() {
 
         Consumer<ServerResponse> consumerSecond = serverResponse -> {
             assertEquals(400, serverResponse.statusCode().value());
+            assertEquals(400, serverResponse.statusCode().value());
```
Comment (And1sS) [inline src/test/java/org/prebid/cache/handlers/PostCacheHandlerTests.java:235]: Redundant check

**Review by And1sS — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/cache/handlers/GetCacheHandler.java (hunk 1 above)
Comment (SerhiiNahornyi) [reply src/main/java/org/prebid/cache/handlers/GetCacheHandler.java:89]: Cache uses <java.version>1.8</java.version>

File: src/main/java/org/prebid/cache/handlers/GetCacheHandler.java (hunk 4 above)
Comment (SerhiiNahornyi) [reply src/main/java/org/prebid/cache/handlers/GetCacheHandler.java:153]: Here with String.format it is more null safe way,
else I should use "" + config.getPrefix() + keyIdParam

**Review by And1sS — APPROVED** [review]

**Comment by SerhiiNahornyi** [comment]
https://github.com/prebid/prebid-server/issues/1620

**Pull request merged**